compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2174,6 +2190,91 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             None
         }
     }
+
+    /// Finds a cfg-ed out item inside `module` with the matching name.
+    #[instrument(skip(self, err))]
+    pub(crate) fn find_cfg_stripped(
+        &mut self,
+        err: &mut Diagnostic,
+        last_segment: &Segment,
+        module: DefId,
+    ) {
+        // crate up:
+        // mod wrong { #[cfg(false)] fn spawn() {} }
+        // mod right { #[cfg(false)] fn spawn() {} }
+        //                              ^^^^^ we want to find this stripped out item
+
+        // this crate:
+        // use up::right;
+        // right::spawn();
+
+        // up:
+        //          crate::wrong::spawn
+        //          ^^^^^^^^^^^^^^^^^^^--cfged_path 0
+        //
+        //          crate::right::spawn
+        //                 ^^^^^--module
+        //          ^^^^^^^^^^^^^^^^^^^--cfged_path 1
+        //
+        // here:
+        //          right::spawn
+        //                 ^^^^^--last_segment
+
+        // We want to skip the first one, because it doesn't match up.
+
+        let module_path = self.tcx.def_path(module);
+
+        debug!(?module_path, "path to parent module of cfged out item");
+
+        let symbols = self.tcx.stripped_out_item_names(module.krate);
+
+        'stripped_items: for (cfged_path, meta_item) in symbols {
+            debug!(?cfged_path, ?meta_item, "Trying next stripped out item");
+
+            let mut module_segments = module_path.data.iter();
+            // Skip the crate root, as it's not present in def_path.
+            let mut cfged_path_segments = cfged_path.iter().skip(1);
+
+            // Ensure that both paths are in the same module. We make sure to not exhaust the `cfged_path_segments`
+            // iterator when `module_segments` is exhausted.
+            while let Some(mod_seg) = module_segments.next() {
+                if let Some(cfged_seg) = cfged_path_segments.next() {
+                    match (mod_seg.data, cfged_seg) {
+                        (DefPathData::TypeNs(sym), cfg_ident) if sym == cfg_ident.name => {}
+                        _ => {
+                            debug!(?mod_seg, ?cfged_seg, "Module path segment doesn't match");
+                            continue 'stripped_items;
+                        }
+                    }
+                } else {
+                    debug!("No more cfged path segments");
+                    continue 'stripped_items;
+                }
+            }
+
+            let Some(cfged_ident) = cfged_path_segments.next() else {
+                debug!("No more cfged path segments");
+                continue;
+            };
+
+            if cfged_ident.name != last_segment.ident.name {
+                debug!("Final cfged out item name doesn't match");
+                continue;
+            }
+
+            err.span_note(
+                cfged_path.last().unwrap().span,
+                &format!("found an item that was configured out"),
```

Comment:
This note needs work.
For example with your uwu example:
```rust
    #[cfg(FALSE)]
    pub fn uwu() {}
```
attempting to use it gives
```rust
note: found an item that was configured out
  --> $DIR/diagnostics-same-crate.rs:3:12
   |
LL |     pub fn uwu() {}
   |            ^^^
```
I think it would be better to say something like:
```
note: crate `foo` has a function named `uwu` but it is inactive because the cfg predicate `FALSE` evaluated to false.
  --> $DIR/diagnostics-same-crate.rs:3:12
   |     #[cfg(FALSE)]
LL |     pub fn uwu() {}
   |            ^^^
```

I prefer to not guess that it actually is what the user needs, so I'd go with "found an item named {name}" rather than "found an item".

Reply:
I went with "note: crate `uwu` has an item named `uwu` but it is inactive because its cfg predicate evaluated to false" for now. I don't want to mention the predicate in the message directly because it could get quite long and the diagnostic already points at it.